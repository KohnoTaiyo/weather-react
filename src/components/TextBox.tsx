import "./TextBox.scss";

type TextBoxProps = {
  value: string;
  onChange: (text: string) => void;
};

export function TextBox(props: TextBoxProps) {
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => props.onChange(e.target.value);

  return (
    <div className="textBox">
      <input
        type="text"
        className="textBox__input"
        placeholder="location"
        value={props.value}
        onChange={handleChange}
        required
      />
      <input type="submit" value="Search" className="textBox__submit" />
    </div>
  );
}
