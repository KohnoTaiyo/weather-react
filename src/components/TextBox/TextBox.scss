@use "../../styles/variables" as *;

.textBox {
  display: flex;
  column-gap: 8px;
  justify-content: space-between;

  &__input {
    width: 100%;
    padding: 8px;
    border: 1px solid $primaryColor;
    border-radius: 4px;
  }

  &__submit {
    padding: 8px;
    border-radius: 4px;
    background-color: $primaryColor;
    color: $whiteColor;
    font-weight: bold;

    &:focus {
      outline: 2px solid $skyBlueColor;
      outline-offset: 1px;
    }
  }
}
