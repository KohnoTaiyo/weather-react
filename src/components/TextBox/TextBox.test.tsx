import { render, screen } from "@testing-library/react";

import { TextBox } from "./TextBox";

const onChangeMock = jest.fn();

describe("TextBox", () => {
  test("renders correctly dom", () => {
    render(<TextBox value="value" onChange={onChangeMock} />);
    expect(screen.getByRole("textbox")).toHaveValue("value");
    expect(screen.getByRole("button", { name: /Search/i })).toBeTruthy();
  });

  test("snapshot", () => {
    const { container } = render(<TextBox value="value" onChange={onChangeMock} />);
    expect(container).toMatchSnapshot();
  });
});
