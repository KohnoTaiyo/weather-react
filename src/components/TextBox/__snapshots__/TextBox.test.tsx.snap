// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox snapshot 1`] = `
<div>
  <div
    class="textBox"
  >
    <input
      class="textBox__input"
      placeholder="location"
      required=""
      type="text"
      value="value"
    />
    <input
      class="textBox__submit"
      tabindex="0"
      type="submit"
      value="Search"
    />
  </div>
</div>
`;
